### ISSUE TITLE
Have a single unix user used for all the thin-edge daemons

### ISSUE THREAD

**Issue by rina23q** [thin-edge/thin-edge.io#1031]
### Discussed in https://github.com/thin-edge/thin-edge.io/discussions/1014

<div type='discussions-op-text'>

<sup>Originally posted by **didier-wenzek** March 17, 2022</sup>
Currently, thin-edge requires the creation of 3 unix users: `tedge`, `tedge-mapper` and `tedge-agent` (plus 3 unix groups with the same names). Is this useful? Why not a single user `tedge` ?

Having distinguished users for all the daemons makes things more complex then necessary: 
* As noted by @PradeepKiruvale, the [initialization of tedge would be simple](https://github.com/thin-edge/thin-edge.io/pull/993#issuecomment-1069232055).
* As noted by @mstoffel-sag, this is useless in a container.
* The security compartmentalization is illusory: the different tedge daemons communicate over MQTT and, once one is compromised, malicious messages can be sent to the others. From a security perspective the agent and the mapper forms a whole.
* The `tedge` user is created only to own files, and not a single process use it.

### Proposal

* Use a single unix user `tedge` and group `tedge` to own all the thin-edge related resources and to run both the mapper and the agent.
* This `tedge` user should not be mandatory but only a proposed default.
  </div>

**Comment by matthiasbeyer** [comment]
May I ask why a dedicated user? Where is the benefit of having a dedicated user for thin-edge?

I feel like 

> This tedge user should not be mandatory but only a proposed default. 

already points into this direction, but to make sure I wanted to ask anyways. I hope you don't mind.

**Comment by didier-wenzek** [comment]
> May I ask why a dedicated user? Where is the benefit of having a dedicated user for thin-edge?

Yes, it would be good to remove this requirement. I didn't not in this issue because fixing this would be by far more involved. We made the mistake to control which user is doing what from *within* the `tedge` command. Launched with `sudo tedge` the process switches to `tedge`, `mosquitto` or `root` depending of the action.

This needs to be fixed, but cannot be done as quickly as removing the two other users.
